File: src/main/org/nlogo/prim/etc/_patchrightandahead.java
package org.nlogo.prim.etc ;

import org.nlogo.api.LogoException;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _patchrightandahead
	extends Reporter
{
	@Override
	public Syntax syntax()
	{
		return Syntax.reporterSyntax
			( new int[] { Syntax.TYPE_NUMBER , Syntax.TYPE_NUMBER } ,
			  Syntax.TYPE_PATCH , "-T--" ) ;
	}
	@Override
	public Object report( final Context context ) 
		throws LogoException
	{
		try
		{
			org.nlogo.agent.Turtle turtle = (org.nlogo.agent.Turtle) context.agent ;
			return turtle.getPatchAtHeadingAndDistance
				( argEvalDoubleValue( context , 0 ) ,
				  argEvalDoubleValue( context , 1 ) ) ;
		}
		catch( org.nlogo.api.AgentException exc )
		{
			return org.nlogo.api.Nobody.NOBODY ;
		}
	}
}

File: src/main/org/nlogo/prim/etc/_patchrightandahead.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.agent.Turtle
import org.nlogo.api.{ Syntax, AgentException, Nobody }
import org.nlogo.nvm.{ Context, Reporter }

@annotation.strictfp
class _patchrightandahead extends Reporter {

  override def syntax =
    Syntax.reporterSyntax(
      Array(Syntax.NumberType, Syntax.NumberType),
      Syntax.PatchType, "-T--")

  override def report(context: Context): AnyRef =
    try
      context.agent.asInstanceOf[Turtle]
        .getPatchAtHeadingAndDistance(
          argEvalDoubleValue(context, 0),
          argEvalDoubleValue(context, 1))
    catch { case _: AgentException => Nobody }

}
